ec/resources/dnsroute/secondary_zone.go

```diff
@@ -111,41 +138,46 @@ func ResourceSecondaryZoneGroup() *schema.Resource {
 	}
 }
 
-func ResourceSecondaryZoneGroupCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	var diags diag.Diagnostics
+func ResourceSecondaryZoneGroupCreate(
+	ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	// Initialize Route DNS Service
 	accountNumber := d.Get("account_number").(string)
-	name := d.Get("name").(string)
-
 	config := m.(**api.ClientConfig)
-	(*config).AccountNumber = accountNumber
+	routeDNSService, err := buildRouteDNSService(**config)
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
 
+	// Construct Secondary Zone Group Object
+	name := d.Get("name").(string)
 	zoneComposition := d.Get("zone_composition").([]interface{})
 	zc := zoneComposition[0].(map[string]interface{})
 	masterGroupID := zc["master_group_id"].(int)
 	zones := zc["zones"].([]interface{})
-	zoneArr := []api.SecondaryZoneRequest{}
+	zoneArr := []routedns.SecondaryZone{}
```

Comment:
Could use an "ExpandSecondaryZone" function and potentially reuse for Update

Reply:
I am not sure why this isn't rendering in the UI diff but if you look at the commit for the PR merged into this branch you'll see the change.

https://github.com/EdgeCast/terraform-provider-edgecast/pull/77/commits/b2404951c10da06dfdcf76a96fca0d1d74b9238a#diff-78bae89ee6828ba869232f4c687f496aa7232775049e04877c318193e85877c4R160

Done! Thank you!